File: app/src/main/java/com/letianpai/robot/time/callback/GeneralInfoCallback.java
package com.letianpai.robot.time.callback;

import com.letianpai.robot.time.parser.general.GeneralInfo;

/**
 * @author liujunbin
 */
public class GeneralInfoCallback {

    private GeneralInfoUpdateListener mGeneralInfoUpdateListener;

    private static class GeneralInfoCallbackHolder {
        private static GeneralInfoCallback instance = new GeneralInfoCallback();
    }

    private GeneralInfoCallback() {

    }

    public static GeneralInfoCallback getInstance() {
        return GeneralInfoCallbackHolder.instance;
    }

    public interface GeneralInfoUpdateListener {
        void onAtCmdResultReturn(GeneralInfo generalInfo);

    }

    public void setGeneralInfoUpdateListener(GeneralInfoUpdateListener listener) {
        this.mGeneralInfoUpdateListener = listener;
    }

    public void setGeneralInfo(GeneralInfo generalInfo) {
        if (mGeneralInfoUpdateListener != null) {
            mGeneralInfoUpdateListener.onAtCmdResultReturn(generalInfo);
        }
    }


}

File: app/src/main/java/com/letianpai/robot/time/callback/GeneralInfoCallback.kt
package com.letianpai.robot.time.callback

import com.letianpai.robot.time.parser.general.GeneralInfo

/**
 * @author liujunbin
 */
class GeneralInfoCallback private constructor() {
    private var mGeneralInfoUpdateListener: GeneralInfoUpdateListener? = null

    private object GeneralInfoCallbackHolder {
        val instance = GeneralInfoCallback()
    }

    fun interface GeneralInfoUpdateListener {
        fun onAtCmdResultReturn(generalInfo: GeneralInfo)
    }

    fun setGeneralInfoUpdateListener(listener: GeneralInfoUpdateListener?) {
        this.mGeneralInfoUpdateListener = listener
    }

    fun setGeneralInfo(generalInfo: GeneralInfo) {
        if (mGeneralInfoUpdateListener != null) {
            mGeneralInfoUpdateListener!!.onAtCmdResultReturn(generalInfo)
        }
    }


    companion object {
        val instance: GeneralInfoCallback
            get() = GeneralInfoCallbackHolder.instance
    }
}
